### PR TITLE
Add fine locations compatibility.

### LINKED ISSUE #2: Error accumulation
Given a list of coordinates with more than 5 decimals the coordinates start to drift ever so slightly during the encode function, resulting in errors that show most notably at small scale and with long lists.

Example:

    decode(encode([[0.000003,0],[0.000008,0],[0.000013,0],[0.000029,0],[0.000045,0],[0.000061,0],[0.000077,0],[0.000093,0]]))

expected result:

    [[0,0],[0.00001,0],[0.00001,0],[0.00003,0],[0.00005,0],[0.00006,0],[0.00008,0],[0.00009,0]]

actual result:

    [[0,0],[0,0],[0,0],[0.00002,0],[0.00004,0],[0.00006,0],[0.00008,0],[0.0001,0]]

I found that the culprit is during the difference computation, which doesn't take into account the desired result after rounding. I could fix the encoding function by replacing lines 8 and 9 with

    str += encode.chars( encode.sign( (Math.round(values[i]*1e5) - Math.round(px*1e5))/1e5 ) )
    str += encode.chars( encode.sign( (Math.round(values[i + 1]*1e5) - Math.round(py*1e5))/1e5 ) )

Although the fix would add some overhead and is not optimized, I'll leave the decision up to the maintainer to further optimize the above code or simply drop it to preserve performance.

### PR DESCRIPTION
**Pull request by Gloix**
This PR fixes #2. Made some changes to core variables for performance, as opposed to the code suggested in that issue. Also, I didn't update the performance values in the Readme since I don't know or probably have at my disposal the original benchmarking setup.

### CODE REVIEW

File: lib/encode.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
 function encode( points ) {
 
   var values = encode.flatten( points )
-  var px = 0, py = 0
+  var px_e5 = 0, py_e5 = 0
```
Comment (jhermsmeier) [inline lib/encode.js]: I'd rather leave the `_e5` suffix off here, and add a comment about preserving floating point precision in the loop below.

File: lib/encode.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
   var str = ''
 
   for( var i = 0; i < values.length; i += 2 ) {
-    str += encode.chars( encode.sign( values[i] - px ) )
-    str += encode.chars( encode.sign( values[i + 1] - py ) )
-    px = values[i]
-    py = values[i + 1]
+    var x_e5 = Math.round( values[i] * 1e5 )
+    var y_e5 = Math.round( values[i + 1] * 1e5 )
```
Comment (jhermsmeier) [inline lib/encode.js]: Same as above here (the `_e5` suffix), and I'd also prefer to have the variable declarations outside the loop (i.e. as `px` and `py` are)

**Comment by Gloix** [comment]
Added a commit to comply with the requested changes. Although the comment was included in the variable definition section to clarify the special usage of those variables. Let me know if you prefer to change that or something else.

**Comment by jhermsmeier** [comment]
Excellent! Thank you again for finding & fixing this! I'll publish a new version asap.

**Pull request merged**